### ISSUE TITLE
If there are too many messages, archive command gets "too many open files"

### ISSUE THREAD

**Issue by tsiemens** [tsiemens/gmail-tools#4]
eg:
```
gmailcli archive -r
Fetching inbox... Loading message details 1502/1502
Failed to get message: Get https://www.googleapis.com/gmail/v1/users/me/messages/16b7f7dcb7a8ec37?alt=json: dial tcp 216.58.217.42:443: socket: too many open files
```

May need to rate limit these.

Current workaround (since there is not max argument for archive) if to either use search + touch, or manually specify the base command with something like `after:2004/04/16`

**Comment by tsiemens** [comment]
Apparently this can also happen:

```Failed to get message: googleapi: Error 429: Too many concurrent requests for user, rateLimitExceeded```

**Comment by tsiemens** [comment]
Fixed in f4566f3
